homeassistant/components/proxmoxve/sensor.py

```diff
@@ -0,0 +1,113 @@
+"""Sensor to read Proxmox VE data."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.const import DATA_GIBIBYTES
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import COORDINATORS, DOMAIN, PROXMOX_CLIENTS, ProxmoxEntity
+
+BYTE_TO_GIBIBYTE = 1.074e9
+
+SENSOR_DESCRIPTIONS = [
+    (
+        SensorEntityDescription(
+            key="mem_gib",
+            name="mem_gib",
+            native_unit_of_measurement=DATA_GIBIBYTES,
+            icon="",
+        ),
+        lambda data: round(int(data.mem) / BYTE_TO_GIBIBYTE, 2),
```

Comment:
Can you move the lambda inside the EntityDescription?

You will probably need to:
- Create a ProxmoxVERequiredKeyMixin class with the definition of the native_lambda
- Create a ProxmoxVESensorEntityDescription which inherits ProxmoxVERequiredKeyMixin and SensorEntityDescription
- Use a class attribute on ProxmoxSensor to indicate the use of the ProxmoxVESensorEntityDescription

Reply:
Good idea. Thanks for all these suggestions btw. Most of this stuff isn't very clear right away. I made the changes in the latest commit. Let me know if there's something wrong with the way I did it.